libr/anal/xrefs.c

```diff
@@ -24,166 +21,117 @@ xrefs
 20: call 10
 #endif
 
+// XXX: is it possible to have multiple type for the same (from, to) pair?
+//      if it is, things need to be adjusted
+
+#define u64_to_key(x) (sdb_fmt ("%"PFMT64x, (x)))
+
+#define ht_find_u64(_ht,_key,_found) (ht_find ((_ht), u64_to_key (_key), (_found)))
+#define ht_insert_u64(_ht,_key,_value) (ht_insert ((_ht), u64_to_key (_key), _value))
+#define ht_update_u64(_ht,_key,_value) (ht_update ((_ht), u64_to_key (_key), _value))
+#define ht_delete_u64(_ht,_key) (ht_delete ((_ht), u64_to_key (_key)))
+
+R_API RAnalRef *r_anal_ref_new() {
+	RAnalRef *ref = R_NEW0 (RAnalRef);
+	if (ref) {
+		ref->addr = -1;
+		ref->at = -1;
+		ref->type = R_ANAL_REF_TYPE_CODE;
+	}
+	return ref;
+}
+
+R_API RList *r_anal_ref_list_new() {
+	return r_list_newf (r_anal_ref_free);
+}
+
+R_API void r_anal_ref_free(void *ref) {
+	free (ref);
+}
+void xrefs_ht_free(HtKv *kv) {
+	free (kv->key);
+	ht_free (kv->value);
+	free (kv);
+}
+
+void xrefs_ref_free(HtKv *kv) {
+	free (kv->key);
+	r_anal_ref_free (kv->value);
+	free (kv);
+}
+
 static const char *analref_toString(RAnalRefType type) {
 	switch (type) {
 	case R_ANAL_REF_TYPE_NULL:
 		/* do nothing */
 		break;
 	case R_ANAL_REF_TYPE_CODE:
-		return "code.jmp";
+		return "code jmp";
```

Comment:
i think we can remove the code and data words in here, because jmp/call only makes sense for code and mem and string only for data., lets rename those enums too.

the problem i see is that TYPE_CODE is used for LEA when pointing to code and JMP, so cant be TYPE_JMP.. what do you think? we can keep those, but maybe remove the code/data to start

Reply:
I'm fine with removing `code` and `data` prefixes. Though I would convert R_ANAL_REF_TYPE_CODE to "code" instead of "jmp", because as you said TYPE_CODE is not only used for jmps.

Reply:
So... weird idea but if instead of having this `analref_toString` function we just use `r_anal_xrefs_type_tostring`? it would be consistent everywhere at least!

Reply:
yeah sounds good to me, if you want to do any other change here we can discuss it after merging. focus on pleasing travis and ill merge :))))